app/assets/javascripts/discourse/app/components/d-modal.gjs

```diff
@@ -77,9 +121,25 @@ export default class DModal extends Component {
       return;
     }
 
-    return this.args.closeModal?.({
-      initiatedBy: CLOSE_INITIATED_BY_CLICK_OUTSIDE,
-    });
+    return this.closeModal(CLOSE_INITIATED_BY_CLICK_OUTSIDE);
+  }
+
+  @action
+  closeModal(initiatedBy) {
+    if (!this.args.closeModal) {
+      return;
+    }
+
+    this.closing = true;
+
+    this.wrapperElement.animate([{ transform: `translateY(100%)` }], {
```

Comment:
Do we need to gate this logic on `site.mobile`?

Reply:
[Kapture 2024-03-21 at 17.52.39.webm](https://github.com/discourse/discourse/assets/339945/dbd540b6-e053-488c-8f53-5bca2eeef122)

Yes I hesitated, I thought it was looking good, but maybe I just focus on mobile for now.